File: Backend/models/event.js
var mongoose = require('mongoose');
var database = require('../database');
var list = require('./list');

var event = new mongoose.Schema({
  EventID: Number,
  members: [{
    _id : false,
    UserId: Number,
    friendlyName: String,
    isAttending: { type: String, enum: ['Invited', 'Attending', 'Declined', 'Left', 'Owner'] },
    Nofications: [{
        itemList: Boolean,
        messageBoard: Boolean
    }]
  }],
  itemList: [list],
  totalEstCost: {type: Number, default: 0},
  totalActCost: {type: Number, default: 0},
  what: String,
  why: String,
  where: String,
  when: String,
  endDate: String,
  picture: String,
  fromTime: String,
  toTime: String
});

module.exports = event;

File: Android/Plan-It/app/src/main/java/com/plan_it/mobile/plan_it/Event.java
package com.plan_it.mobile.plan_it;

import android.graphics.Bitmap;

/**
 * Created by Kevin on 17-Oct-2015.
 */
enum IsAttending{INVITED, ATTENDING, DECLINED, LEFT, OWNER}
public class Event {
    int eventID;
    String name;
    String owner;
    String description;
    String location;
    String fromDate;
    String toDate;
    String fromTime;
    String toTime;
    Bitmap photoId;
    IsAttending isAttending;
    boolean itemList;
    boolean messageBoard;

    Event(int eventID,String name, String owner, String description,String location, Bitmap photoId, String fromDate, String toDate, String fromTime, String toTime, IsAttending isAttending, boolean itemList, boolean messageBoard) {
        this.eventID = eventID;
        this.name = name;
        this.owner = owner;
        this.description = description;
        this.location = location;
        this.photoId = photoId;
        this.fromDate = fromDate;
        this.toDate = toDate;
        this.toTime = toTime;
        this.fromTime = fromTime;
        this.isAttending = isAttending;
        this.itemList = itemList;
        this.messageBoard = messageBoard;
    }
}
